MySQL database doesn't reconnect after connection failure

**LWC Version (/lwc admin version):** 2.2.4-a88d70a (I know this isn't latest but I don't see anything related to MySQL connection changes)
**Server version (/version):** git-Origami-"6593510" (basically Paper 1.15.2 b371)

**Short five word summary of issue:** MySQL doesn't reconnect

**Detailed description of problem:**

When the connection to the MySQL fails (e.g. due to a restart or network issues) the plugin does not auto-reconnect and throws errors on any protectioni nteraction.

**If there was an error in console, put it here:**
```
[00:36:28] [Server thread/ERROR]: Could not pass event CreatureSpawnEvent to LWC v2.2.4-a88d70a
com.griefcraft.scripting.ModuleException: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure

The last packet successfully received from the server was 3,735,612 milliseconds ago.  The last packet sent successfully to the server was 3,447,243 milliseconds ago.
        at com.griefcraft.sql.Database.printException(Database.java:198) ~[?:?]
        at com.griefcraft.sql.PhysDB.resolveProtections(PhysDB.java:861) ~[?:?]
        at com.griefcraft.sql.PhysDB.resolveProtection(PhysDB.java:881) ~[?:?]
        at com.griefcraft.sql.PhysDB.loadProtection(PhysDB.java:972) ~[?:?]
        at com.griefcraft.sql.PhysDB.loadProtection(PhysDB.java:934) ~[?:?]
        at com.griefcraft.lwc.LWC.findProtection(LWC.java:1362) ~[?:?]
        at com.griefcraft.lwc.LWC.findProtection(LWC.java:1345) ~[?:?]
        at com.griefcraft.listeners.LWCEntityListener.entityCreatedByPlayer(LWCEntityListener.java:216) ~[?:?]
        at com.griefcraft.listeners.LWCEntityListener.onCreateSpawn(LWCEntityListener.java:199) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor41.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[craftbukkit.jar:git-Origami-"6593510"]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[craftbukkit.jar:git-Origami-"6593510"]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[craftbukkit.jar:git-Origami-"6593510"]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[craftbukkit.jar:git-Origami-"6593510"]
        at org.bukkit.craftbukkit.v1_15_R1.event.CraftEventFactory.callCreatureSpawnEvent(CraftEventFactory.java:639) ~[craftbukkit.jar:git-Origami-"6593510"]
        at org.bukkit.craftbukkit.v1_15_R1.event.CraftEventFactory.doEntityAddEventCalling(CraftEventFactory.java:560) ~[craftbukkit.jar:git-Origami-"6593510"]
        at net.minecraft.server.v1_15_R1.WorldServer.addEntity0(WorldServer.java:1267) ~[craftbukkit.jar:git-Origami-"6593510"]
        at net.minecraft.server.v1_15_R1.WorldServer.addEntity(WorldServer.java:1175) ~[craftbukkit.jar:git-Origami-"6593510"]
        at org.bukkit.craftbukkit.v1_15_R1.CraftWorld.addEntity(CraftWorld.java:1965) ~[craftbukkit.jar:git-Origami-"6593510"]
        at org.bukkit.craftbukkit.v1_15_R1.CraftWorld.spawn(CraftWorld.java:1972) ~[craftbukkit.jar:git-Origami-"6593510"]
        at org.bukkit.craftbukkit.v1_15_R1.CraftWorld.spawn(CraftWorld.java:1556) ~[craftbukkit.jar:git-Origami-"6593510"]
        at org.bukkit.craftbukkit.v1_15_R1.CraftWorld.spawnEntity(CraftWorld.java:782) ~[craftbukkit.jar:git-Origami-"6593510"]
        at io.github.apfelcreme.MbPetsNoLD.Pet.Pet.lambda$spawn$0(Pet.java:361) ~[?:?]
        at org.bukkit.craftbukkit.v1_15_R1.scheduler.CraftTask.run(CraftTask.java:99) ~[craftbukkit.jar:git-Origami-"6593510"]
        at org.bukkit.craftbukkit.v1_15_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[craftbukkit.jar:git-Origami-"6593510"]
        at net.minecraft.server.v1_15_R1.MinecraftServer.b(MinecraftServer.java:1286) ~[craftbukkit.jar:git-Origami-"6593510"]
        at net.minecraft.server.v1_15_R1.DedicatedServer.b(DedicatedServer.java:432) ~[craftbukkit.jar:git-Origami-"6593510"]
        at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:1203) ~[craftbukkit.jar:git-Origami-"6593510"]
        at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:992) ~[craftbukkit.jar:git-Origami-"6593510"]
        at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure

The last packet successfully received from the server was 3,735,612 milliseconds ago.  The last packet sent successfully to the server was 3,447,243 milliseconds ago.
        at jdk.internal.reflect.GeneratedConstructorAccessor227.newInstance(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
        at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:481) ~[?:?]
        at com.mysql.jdbc.Util.handleNewInstance(Util.java:425) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:990) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2512) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2675) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2465) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1912) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.PreparedStatement.executeQuery(PreparedStatement.java:2020) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.griefcraft.sql.PhysDB.resolveProtections(PhysDB.java:858) ~[?:?]
        ... 28 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.io.InputStream.available()" because "this.mysqlInput" is null
        at com.mysql.jdbc.MysqlIO.clearInputStream(MysqlIO.java:892) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2471) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2675) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2465) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1912) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.mysql.jdbc.PreparedStatement.executeQuery(PreparedStatement.java:2020) ~[craftbukkit.jar:git-Origami-"6593510"]
        at com.griefcraft.sql.PhysDB.resolveProtections(PhysDB.java:858) ~[?:?]
        ... 28 more

```

--- a/src/main/java/com/griefcraft/sql/Database.java
+++ b/src/main/java/com/griefcraft/sql/Database.java
@@ -202,7 +202,7 @@ public abstract class Database {
      * @return if the connection was successful
      */
     public boolean connect() throws Exception {
-        if (connection != null) {
+        if (connection != null && !connection.isClosed() && connection.isValid(1)) {
             return true;
         }
 
@@ -242,7 +242,6 @@ public abstract class Database {
             connection = driver.connect("jdbc:" + currentType.toString().toLowerCase() + ":" + getDatabasePath(),
                     properties);
             connected = true;
-//			setAutoCommit(true);
             return true;
         } catch (SQLException e) {
             log("Failed to connect to " + currentType + ": " + e.getErrorCode() + " - " + e.getMessage());

